sem/database.py

```diff
@@ -329,16 +366,27 @@ def get_results(self, params=None, result_id=None):
             else:
                 query_params[key] = params[key]
 
-        # Handle case where query params has no keys
+        # Handle case where query params has no keys                                                            #TODO - How will this case occur?
         if not query_params.keys():
             return [dict(i) for i in self.db.table('results').all()]
 
         # Create the TinyDB query
         # In the docstring example above, this is equivalent to:
         # AND(OR(param1 == value1), OR(param2 == value2, param2 == value3))
-        query = reduce(and_, [reduce(or_, [
-            where('params')[key] == v for v in value]) for key, value in
-                              query_params.items()])
+    
+        if log_component is None:                                                                                #TODO - form a query of the sort AND(OR(param1 == value1), OR(param2 == value2, param2 == value3),AND(log_component==None))
+            query = reduce(and_, [reduce(or_, [
+                (where('meta')['log_component']==None) and (where('params')[key] == v for v in value)]) for key, value in
```

Comment:
I need to check this more thoroughly, and don't have much time right now - as far as resources go, [this page](https://tinydb.readthedocs.io/en/latest/usage.html#queries) in the TinyDB docs seems to contain some hints. I think having an example script that you can execute, print out the query and output from the db, and play around with could be useful for debugging and getting this right. 

I'll see whether I can tackle this between today and tomorrow!

```suggestion
                (where('meta')['log_component'] is None) and (where('params')[key] == v for v in value)]) for key, value in
```